test/cpp/asyncprogressqueueworkerstream.cpp

```diff
@@ -0,0 +1,80 @@
+/*********************************************************************
+ * NAN - Native Abstractions for Node.js
+ *
+ * Copyright (c) 2017 NAN contributors
+ *
+ * MIT License <https://github.com/nodejs/nan/blob/master/LICENSE.md>
+ ********************************************************************/
+
+#include <nan.h>
+
+using namespace Nan;  // NOLINT(build/namespaces)
+
+// Custom data type: This serves as an example of how external
+// libraries could be hooked in, populate their objects and send them to JS.
+struct data_t {
+  int index;
+  int data;
+};
+
+template<typename T>
+class ProgressQueueWorker : public AsyncProgressQueueWorker<T> {
+ public:
+  ProgressQueueWorker(
+      Callback *callback
+    , Callback *progress
+    , int iters)
+    : AsyncProgressQueueWorker<T>(callback), progress(progress)
+    , iters(iters) {}
+
+  ~ProgressQueueWorker() {
+    delete progress;
+  }
+
+  void Execute (
+    const typename AsyncProgressQueueWorker<T>::ExecutionProgress& progress) {
+    data_t data;
+    for (int i = 0; i < iters; ++i) {
+      data.index = i;
+      data.data = i * 2;
+      progress.Send(&data, 1);
+    }
+  }
+
+  void HandleProgressCallback(const T *data, size_t count) {
+    HandleScope scope;
+    v8::Local<v8::Object> obj = Nan::New<v8::Object>();
+    Nan::Set(
+      obj,
+      Nan::New("index").ToLocalChecked(),
+      New<v8::Integer>(data->index));
+    Nan::Set(
+      obj,
+      Nan::New("data").ToLocalChecked(),
+      New<v8::Integer>(data->data));
+
+    v8::Local<v8::Value> argv[] = { obj };
+    progress->Call(1, argv);
+  }
+
+ private:
+  Callback *progress;
+  int iters;
+};
+
+NAN_METHOD(DoProgress) {
+  Callback *progress = new Callback(info[1].As<v8::Function>());
+  Callback *callback = new Callback(info[2].As<v8::Function>());
+  AsyncQueueWorker(new ProgressQueueWorker<data_t>(
+      callback
+    , progress
+    , To<uint32_t>(info[0]).FromJust()));
+}
+
+NAN_MODULE_INIT(Init) {
+  Set(target
+    , New<v8::String>("a").ToLocalChecked()
```

Comment:
`"doProgress"`?

Reply:
Again, since it's copied from the existing tests, I don't think we should change it in this PR.   But wouldn't be a bad cleanup afterwards.   I'd make the change now if it wasn't already in all the existing tests (...or if you ask me to do so)   ...   If we change it here, we'll have to change it everywhere else, too:
```
mk@vujade:~/git/nan$ git grep "\"a\"" test/cpp/*
test/cpp/asyncprogressqueueworker.cpp:    , New<v8::String>("a").ToLocalChecked()
test/cpp/asyncprogressworker.cpp:    , New<v8::String>("a").ToLocalChecked()
test/cpp/asyncprogressworkersignal.cpp:    , New<v8::String>("a").ToLocalChecked()
test/cpp/asyncprogressworkerstream.cpp:    , New<v8::String>("a").ToLocalChecked()
test/cpp/asyncworker.cpp:    , New<v8::String>("a").ToLocalChecked()
test/cpp/asyncworkererror.cpp:    , New("a").ToLocalChecked()
test/cpp/bufferworkerpersistent.cpp:    , New<v8::String>("a").ToLocalChecked()
test/cpp/setcallhandler.cpp:    , New("a").ToLocalChecked()
```

Reply:
As much as "doProgress" would make sense here, I think uniformity amongst all of the tests is most important, and it's kinda nice how all of the tests export the addon function using the same name, `a` , but I'll agree that if I were writing new code today I would use a meaningful name.   If you want me to change it, I will... but I think it should remain, at least for this pull request.   If we want to make them meaningful names then lets chaneg them all at once after this is merged...  but I repeat, I am happy to comply with whatever you'd prefer.

Reply:
On a second look, I realize that yes, there are some tests that call the addon `"a"` while others do have meaningful names.  So, I've changed my mind.  Renamed `"a"` to  `"doProgress"` in 0c0248ab1e62cd2c9b0d900a7113a194d1bf1ed7  ...  We should probably clean up those others at some point as well.